src/Storages/MergeTree/MergeTreePartitionCompatibilityVerifier.cpp

```diff
@@ -0,0 +1,119 @@
+#include <Storages/MergeTree/MergeTreePartitionCompatibilityVerifier.h>
+#include <Storages/MergeTree/MergeTreeData.h>
+#include <Storages/MergeTree/MergeTreePartitionGlobalMinMaxIdxCalculator.h>
+#include <Interpreters/MonotonicityCheckVisitor.h>
+#include <Interpreters/getTableExpressions.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+    bool isDestinationPartitionExpressionMonotonicallyIncreasing(
+        const std::vector<Range> & hyperrectangle,
+        const MergeTreeData & destination_storage
+    )
+    {
+        auto destination_table_metadata = destination_storage.getInMemoryMetadataPtr();
+
+        auto key_description = destination_table_metadata->getPartitionKey();
+        auto definition_ast = key_description.definition_ast->clone();
+
+        auto table_identifier = std::make_shared<ASTIdentifier>(destination_storage.getStorageID().getTableName());
+        auto table_with_columns = TableWithColumnNamesAndTypes {
+            DatabaseAndTableWithAlias(table_identifier, {}),
+            destination_table_metadata->getColumns().getOrdinary()
+        };
+
+        auto expression_list = extractKeyExpressionList(definition_ast);
+
+        Monotonicity monotonicity;
+
+        for (auto i = 0u; i < expression_list->children.size(); i++)
+        {
+            MonotonicityCheckVisitor::Data data {{table_with_columns}, destination_storage.getContext(), {}};
+
+            data.range = hyperrectangle[i];
+
+            MonotonicityCheckVisitor(data).visit(expression_list->children[i]);
```

Comment:
After modifications suggested in `MonotonicityCheckVisitor.h`:
`visit(expression_list->children[i], hyperrectangle[i]);`

Reply:
Outdated, right?